src/features/governance/hooks/useGovernanceProposals.test.ts

```diff
@@ -0,0 +1,167 @@
+      ...(await importActual()),
+    }));
+  });
+
+  it('maps celoscan events to proposal ids', async () => {
+    vi.mock('src/features/explorers/celoscan', async (importActual) => ({
+      ...(await importActual()),
+      queryCeloscanLogs: () =>
+        Promise.resolve([
+          { topics: [`0x`, `0x1`] },
+          { topics: [`0x`, `0x2`] },
+          { topics: [`0x`, `0x3`] },
+          { topics: [`0x`, `0x10`] },
+        ] as unknown as TransactionLog[]),
+    }));
+    expect(fetchExecutedProposalIds()).resolves.toMatchInlineSnapshot(`
+      [
+        1,
+        2,
+        3,
+        16,
+      ]
+    `);
+  });
+});
+
+describe('fetchGovernanceMetadata', () => {
+  test('parses the json cache', async () => {
+    // we're not really interesting by debug logs here and we are testing the
+    // parsing functionality in `fetchFromRepository.test.ts`
+    vi.spyOn(logger, 'debug').mockImplementation(() => {});
+
+    (await fetchGovernanceMetadata()).forEach((cachedProposalFromRepo) => {
+      expect(cachedProposalFromRepo).toHaveProperty('stage');
+      expect(cachedProposalFromRepo).toHaveProperty('cgp');
+      expect(cachedProposalFromRepo).toHaveProperty('cgpUrl');
+      expect(cachedProposalFromRepo).toHaveProperty('cgpUrlRaw');
+      expect(cachedProposalFromRepo).toHaveProperty('title');
+      expect(cachedProposalFromRepo).toHaveProperty('author');
+    });
+  });
+});
+
+describe('getExpiryTimestamp', () => {
+  test('Approval', () => {
+    expect(getExpiryTimestamp(ProposalStage.Approval, 1000)).toMatchInlineSnapshot(`86401000`);
+  });
+  test('Queued', () => {
+    expect(getExpiryTimestamp(ProposalStage.Queued, 1000)).toMatchInlineSnapshot(`2419201000`);
+  });
+  test('Referendum', () => {
+    expect(getExpiryTimestamp(ProposalStage.Referendum, 1000)).toMatchInlineSnapshot(`604801000`);
+  });
+  test('Execution', () => {
+    expect(getExpiryTimestamp(ProposalStage.Execution, 1000)).toMatchInlineSnapshot(`864001000`);
+  });
+  test('Others', () => {
+    [
+      ProposalStage.None,
+      ProposalStage.Expiration,
+      ProposalStage.Rejected,
+      ProposalStage.Withdrawn,
+      ProposalStage.Executed,
+    ].forEach((stage) => {
+      expect(getExpiryTimestamp(stage, 1000)).toBeUndefined();
+    });
+  });
+});
+
+describe('fetchGovernanceProposals', () => {
+  // Know block, mento proposal creation, for abritary reasons
+  process.env.NEXT_PUBLIC_FORK_BLOCK_NUMBER = '0x1baa98c';
+  test('fetched queued and dequeued proposals', async () => {
+    const proposals = await fetchGovernanceProposals(publicClient);
+    expect(proposals).toMatchSnapshot();
+  });
+}, 15_000);
+
+describe('mergeProposalsWithMetadata', () => {
+  test('merge properly', async () => {
+    // make sure we used cached data for tests
+    vi.spyOn(globalThis, 'fetch').mockImplementationOnce(() => {
+      throw new Error('Test error: 403 rate limit exceeded');
+    });
```

Comment:
seems weird.

Reply:
i see why now from above. 

i think best would be to you use guessed it MSW. but like many suggestions its not one i would block merging on